docs/coding-guidelines/add-types-to-SerDe.md

```diff
@@ -0,0 +1,155 @@
+        SerDe.Write(destination, argByteArray);
+        break;
+
+```
+
+Typically the type identifier is written, then the length and then the data. Once you have written 
+the code that will perform the write you need to add a case to `GetTypeId` for your new identifier:
+
+```csharp
+
+    case TypeCode.Double:
+        return s_doubleTypeId;
+
+```
+
+The next step is to add the derserialization code to scala (read code).
+
+In src/main/scala/org/apache/spark/api/dotnet/SerDe.scala (These are versioned so you may need to 
+try and find it). In the method `readTypedObject` add a new case statement for your new type 
+identifier:
+
+```scala
+
+def readTypedObject(dis: DataInputStream, dataType: Char): Object = {
+    dataType match {
+      case 'n' => null
+      case 'i' => new java.lang.Integer(readInt(dis))
+      case 'g' => new java.lang.Long(readLong(dis))
+      case 'd' => new java.lang.Double(readDouble(dis))
+      case 'b' => new java.lang.Boolean(readBoolean(dis))
+      case 'c' => readString(dis)
+      case 'e' => readMap(dis)
+      case 'r' => readBytes(dis)
+      case 'l' => readList(dis)
+      case 'D' => readDate(dis)
+      case 't' => readTime(dis)
+      case 'j' => JVMObjectTracker.getObject(readString(dis))
+      case 'R' => readRowArr(dis)
+      case _ => throw new IllegalArgumentException(s"Invalid type $dataType")
+    }
+  }
+
+```
+
+Once you have read the type, you can implement your method to read the data you wrote in csharp, examples:
+
+```scala
+
+  def readBoolean(in: DataInputStream): Boolean = {
+    in.readBoolean()
+  }
+
+  def readRow(in: DataInputStream): Row = {
+    val len = readInt(in)
+    Row.fromSeq((0 until len).map(_ => readObject(in)))
+  }
+
+  def readBytesArr(in: DataInputStream): Array[Array[Byte]] = {
+    val len = readInt(in)
+    (0 until len).map(_ => readBytes(in)).toArray
+  }
+
+```
+
+That should be everything you need to be able to send a new type from the CLR to the JVM but we now
+ need to write back from the JVM to the CLR.
+
+In src/main/scala/org/apache/spark/api/dotnet/SerDe.scala the function `writeObject` contains a 
+switch for each type, a couple of examples:
+
+```scala
+
+    case "int" | "java.lang.Integer" =>
+        writeType(dos, "integer")
+        writeInt(dos, value.asInstanceOf[Int])
+
+    case "java.sql.Timestamp" =>
+          writeType(dos, "time")
+          writeTime(dos, value.asInstanceOf[Timestamp])
+
+    case "[D" =>
+          writeType(dos, "list")
+          writeDoubleArr(dos, value.asInstanceOf[Array[Double]])
+
+```
+
+```scala
+
+ def writeInt(out: DataOutputStream, value: Int): Unit = {
+    out.writeInt(value)
+  }
+
+ def writeTime(out: DataOutputStream, value: Time): Unit = {
+    out.writeDouble(value.getTime.toDouble / 1000.0)
+  }
+
+ def writeDoubleArr(out: DataOutputStream, value: Array[Double]): Unit = {
+    writeType(out, "double")
+    out.writeInt(value.length)
+    value.foreach(v => out.writeDouble(v))
+  }
+
+```
```

Comment:
Can you please show one example to be consistent with E2E example? e.g. `DoubleArrayArray`.